core/src/main/java/org/jruby/RubyThread.java

```diff
@@ -1115,6 +1103,10 @@ public synchronized IRubyObject inspect() {
         return getRuntime().newString(part.toString());
     }
 
+    private boolean notEmpty(CharSequence str) {
+        return str != null && str.toString().length() > 0;
```

Comment:
`str.length()` should do (no need to convert a `RubyString` to a Java `String` here)